libc/include/llvm-libc-macros/limits-macros.h

```diff
@@ -0,0 +1,247 @@
+//===-- Definition of macros from limits.h --------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __LLVM_LIBC_MACROS_LIMITS_MACROS_H
+#define __LLVM_LIBC_MACROS_LIMITS_MACROS_H
+
+#if __has_include_next(<limits.h>)
+
+// Suppress `#include_next is a language extension` warnings.
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wgnu-include-next"
+#elif defined(__GNUC__) // gcc
+#if !defined _GCC_LIMITS_H_
+#define _GCC_LIMITS_H_
```

Comment:
Needs a comment explaining what's going on with GCC's limits.h trying to include_next limits.h.

Reply:
It might be cleaner to move this into its own guard with check, rather than keeping in with the pragmas.

Example:
```c
#if defined(__GNUC__) && !defined(__clang__) && !defined(_GCC_LIMITS_H_)
// comment why
#define _GCC_LIMITS_H_
#endif

#ifdef __clang__
#pragma clang diagnostic push
...
```

Reply:
Done.